prover/src/lib.rs

```diff
@@ -161,30 +162,18 @@ pub trait Prover {
         // figure out which version of the generic proof generation procedure to run. this is a sort
         // of static dispatch for selecting two generic parameter: extension field and hash function.
         match self.options().field_extension() {
-            FieldExtension::None => match self.options().hash_fn() {
-                HashFunction::Blake3_256 => self.generate_proof::<Self::BaseField, Blake3_256<Self::BaseField>>(trace),
-                HashFunction::Blake3_192 => self.generate_proof::<Self::BaseField, Blake3_192<Self::BaseField>>(trace),
-                HashFunction::Sha3_256 => self.generate_proof::<Self::BaseField, Sha3_256<Self::BaseField>>(trace),
-            },
+            FieldExtension::None => self.generate_proof::<Self::BaseField, Self::HashFn>(trace),
```

Comment:
We probably don't need to pass `Self::HashFn` to `self.generate_proof()` as we can access `Self::HashFn` from within `self.generate_proof()`.

Reply:
thanks!